lib/coinbase/smart_contract.rb

```diff
@@ -126,6 +132,95 @@ def self.create_multi_token_contract(
       new(contract)
     end
 
+    # Reads data from a deployed smart contract.
+    #
+    # @param network [Coinbase::Network, Symbol] The Network or Network ID of the Asset
+    # @param contract_address [String] The address of the deployed contract
+    # @param method [String] The name of the method to call on the contract
+    # @param abi [Array, nil] The ABI of the contract. If nil, the method will attempt to use a cached ABI
+    # @param args [Hash] The arguments to pass to the contract method.
+    #   The keys should be the argument names, and the values should be the argument values.
+    # @return [Object] The result of the contract call, converted to an appropriate Ruby type
+    # @raise [Coinbase::ApiError] If there's an error in the API call
+    def self.read(
+      network:,
+      contract_address:,
+      method:,
+      abi: nil,
+      args: {}
+    )
+      network = Coinbase::Network.from_id(network)
+      
+      response = Coinbase.call_api do
+        smart_contracts_api.read_contract(
+          network.normalized_id,
+          contract_address,
+          {
+              method: method,
+              args: args.to_json
+              abi: abi.nil? ? nil : abi.to_json
```

Comment:
```suggestion
              method: method,
              args: args.to_json
              abi: abi.nil? ? nil : abi.to_json
```

This should return a linting error, but the spacing here looks off. Should be 2 spaces per indent